File: app/src/main/java/indi/yume/tools/fragmentmanager/SingleBaseActivity.java
package indi.yume.tools.fragmentmanager;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;

import java.util.HashMap;
import java.util.Map;

/**
 * Created by yume on 15/11/10.
 */
public abstract class SingleBaseActivity extends BaseFragmentManagerActivity {
    protected static final String SINGLE_TAG = "tag";

    private Map<String, Class<?>> map = new HashMap<>();

    @Override
    public Map<String, Class<?>> BaseFragmentWithTag() {
        return map;
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        String className = getIntent().getStringExtra(INTENT_KEY_FRAGMENT_CLASS);
        try {
            map.put(SINGLE_TAG, Class.forName(className));
        } catch (ClassNotFoundException e) {
            e.printStackTrace();
        }
        super.onCreate(savedInstanceState);
        switchToStackByTag(SINGLE_TAG);
    }

    private static final String INTENT_KEY_FRAGMENT_CLASS = "fragmentClass";
    public static Intent createIntent(Context context, Class<?> clazz, Class<? extends SingleBaseActivity> activityClazz){
        Intent intent = new Intent(context, activityClazz);
        intent.putExtra(INTENT_KEY_FRAGMENT_CLASS, clazz.getName());
        return intent;
    }

    public static Intent createIntent(Context context, Class<?> clazz, Class<? extends SingleBaseActivity> activityClazz, Intent intent){
        intent.setClass(context, activityClazz);
        intent.putExtra(INTENT_KEY_FRAGMENT_CLASS, clazz.getName());
        return intent;
    }
}

File: app/src/main/kotlin/indi/yume/tools/fragmentmanager/SingleBaseActivity.kt
package indi.yume.tools.fragmentmanager

import android.content.Context
import android.content.Intent
import android.os.Bundle
import indi.yume.tools.fragmentmanager.event.SwitchAction
import indi.yume.tools.fragmentmanager.model.ItemState
import io.reactivex.Completable

/**
 * Created by yume on 18-3-14.
 */

abstract class SingleBaseActivity : DefaultAppCompatActivity() {

    override val baseFragmentWithTag: Map<String, FragmentCreator> = emptyMap()


    companion object {
        protected const val SINGLE_TAG = "tag"

        fun startSingleActivity(context: Context,
                                item: ItemState,
                                activityClazz: Class<out SingleBaseActivity>,
                                intent: Intent? = null,
                                startActivity: (Intent) -> Unit): Completable {
            return indi.yume.tools.fragmentmanager.startActivity(
                    intent?.setClass(context, activityClazz) ?: Intent(context, activityClazz),
                    startActivity)
                    .flatMap {
                        return@flatMap ApplicationStore.stackManager.dispatch(
                                SwitchAction(targetStack = SINGLE_TAG,
                                        defaultItem = item.copy(stackTag = SINGLE_TAG)))
                    }.toCompletable()
        }
    }
}